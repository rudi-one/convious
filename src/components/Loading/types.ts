export interface LoadingProps {
  mode?: string;
}
