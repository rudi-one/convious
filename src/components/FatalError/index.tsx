import React, {memo} from 'react';
import {Text, View} from 'react-native';
import stylesCommon from 'src/common/style';
import SafeArea from 'src/components/SafeArea';

const FatalErrorComponent: React.FC = () => {
  return (
    <SafeArea>
      <View style={stylesCommon.container}>
        <Text>Critical error</Text>
        <Text>Please reload the application</Text>
        <Text style={{paddingVertical: 20}}>
          “Big things have small beginnings.” ~ Promethus
        </Text>
      </View>
    </SafeArea>
  );
};

export const FatalError = memo(FatalErrorComponent);
