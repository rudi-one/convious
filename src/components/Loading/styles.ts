import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  absolute: {
    position: 'absolute',
    width: '100%',
    height: '100%',
    //    backgroundColor: Colors.grey100,
    zIndex: 10,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default styles;
