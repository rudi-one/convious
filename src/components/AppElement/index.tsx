import {useNavigation} from '@react-navigation/native';
import {NavigationProp, RouteNames} from 'navigation/types';
import React, {memo} from 'react';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import AppProps from 'src/components/AppElement/types';
import styles from './styles';

const AppElementComponent = ({app}: AppProps) => {
  const navigation = useNavigation<NavigationProp>();

  return (
    <TouchableOpacity
      style={styles.container}
      onPress={() => {
        navigation.navigate(RouteNames.app, {app: app});
      }}>
      <View style={styles.imageWrapper}>
        <Image
          source={{
            uri: app.artworkUrl512,
          }}
          style={styles.image}
        />
      </View>
      <View style={styles.information}>
        <Text style={styles.genre}>{app.primaryGenreName}</Text>
        <Text style={styles.title}>{app.trackName}</Text>
      </View>
    </TouchableOpacity>
  );
};

export const AppElement = memo(AppElementComponent);
