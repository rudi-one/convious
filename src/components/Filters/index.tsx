import * as React from 'react';
import {Dispatch, memo, SetStateAction} from 'react';
import {Pressable, ScrollView, View} from 'react-native';
import {Button, Checkbox, Colors, Dialog, Text} from 'react-native-paper';
import {Filter} from 'src/components/Filters/types';
import styles from './styles';

interface FiltersProps {
  visible: boolean;
  setVisible: Dispatch<SetStateAction<boolean>>;
  filters: Filter[];
  filter: (item: Filter) => void;
  clear: (checked: boolean) => void;
}

const FiltersComponent = ({
  visible,
  setVisible,
  filters,
  filter,
  clear,
}: FiltersProps) => {
  const hideDialog = () => setVisible(false);

  return (
    <Dialog visible={visible} onDismiss={hideDialog} style={styles.dialog}>
      <Dialog.Title>Filters (click to filter)</Dialog.Title>
      <Dialog.ScrollArea>
        <ScrollView contentContainerStyle={{paddingHorizontal: 24}}>
          {filters?.map(item => (
            <Pressable
              onPress={() => filter({name: item.name, checked: !item.checked})}
              key={item.name}>
              <View style={styles.checkboxWrapper}>
                <Text style={styles.checkboxLabel}>{item.name}</Text>
                <Checkbox
                  status={item.checked ? 'checked' : 'unchecked'}
                  color={Colors.greenA700}
                  uncheckedColor={Colors.red500}
                />
              </View>
            </Pressable>
          ))}
        </ScrollView>
      </Dialog.ScrollArea>
      <Dialog.Actions>
        <Button onPress={() => clear(false)}>Clear</Button>
        <Button onPress={() => clear(true)}>Select all</Button>
        <Button onPress={hideDialog}>Done</Button>
      </Dialog.Actions>
    </Dialog>
  );
};

export const Filters = memo(FiltersComponent);
