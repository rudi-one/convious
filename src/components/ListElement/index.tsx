import React, {memo} from 'react';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import ElementProps from 'src/components/ListElement/types';
import styles from './styles';

const ListElementComponent = ({
  item,
  imageUrl,
  showArtist,
  navigate,
}: ElementProps) => {
  return (
    <TouchableOpacity style={styles.container} onPress={navigate}>
      <View style={styles.imageWrapper}>
        <Image
          source={{
            uri: imageUrl,
          }}
          style={styles.image}
        />
      </View>
      <View style={styles.information}>
        <Text style={styles.genre}>{item.primaryGenreName}</Text>
        <Text style={styles.title}>
          {showArtist
            ? `${item.artistName}: ${item.trackName}`
            : item.trackName}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

export const ListElement = memo(ListElementComponent);
