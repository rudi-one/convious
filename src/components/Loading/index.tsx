import React, {memo} from 'react';
import {ActivityIndicator, View} from 'react-native';
import {Colors} from 'react-native-paper';
import stylesCommon from 'src/common/style';
import {LoadingProps} from 'src/components/Loading/types';
import styles from './styles';

const LoadingComponent = ({mode}: LoadingProps) => {
  return (
    <View
      style={[
        stylesCommon.container,
        mode === 'absolute' ? styles.absolute : null,
      ]}>
      <ActivityIndicator
        animating={true}
        color={Colors.red800}
        size={'large'}
      />
    </View>
  );
};
export const Loading = memo(LoadingComponent);
