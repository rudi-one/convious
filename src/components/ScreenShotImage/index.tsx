import React, {memo} from 'react';
import {Image, View} from 'react-native';
import styles from './styles';
import {ScreenShotProp} from './types';

const ScreenShotImageComponent = ({item}: {item: ScreenShotProp}) => {
  return (
    <View style={styles.screenshotContainer}>
      <Image
        source={{
          uri: item.url,
        }}
        style={styles.screenshot}
      />
    </View>
  );
};

export const ScreenShotImage = memo(ScreenShotImageComponent);
