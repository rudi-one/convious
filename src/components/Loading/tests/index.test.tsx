import {render} from '@testing-library/react-native';
import React from 'react';
import {Loading} from 'src/components/Loading';

describe('<Loading /> tests', () => {
  it('should match Loading component snapshot', () => {
    const component = <Loading />;

    expect(render(component)).toMatchSnapshot();
  });
  it('should match Loading component snapshot with property', () => {
    const component = <Loading mode={'absolute'} />;

    expect(render(component)).toMatchSnapshot();
  });
});
