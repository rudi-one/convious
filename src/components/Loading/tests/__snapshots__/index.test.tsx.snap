// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> tests should match Loading component snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 27.5,
      },
      null,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#c62828"
    size="large"
  />
</View>
`;

exports[`<Loading /> tests should match Loading component snapshot with property 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 27.5,
      },
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
        "zIndex": 10,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#c62828"
    size="large"
  />
</View>
`;
